### PR TITLE
TT1 to TT3/MSPA 704 sequence identifier

### PR DESCRIPTION
**Pull request by nigelmegitt**
Add EBU-TT 1 to EBU-TT 3 conversion

Does not handle time conversion.
Sets `ebuttp:sequenceIdentifier` to the value of `tt/head/metadata/documentMetadata/documentIdentifier` if present, otherwise uses "TestConverter".
Resets the `conformsToStandard` to say it is EBU-TT-3 conformant.
Sets the `timeBase` to `media` whether you like it or not, but doesn't do any other conversions.

Tests for valid EBU-TT-3 document on output.

### CODE REVIEW

**Comment by nigelmegitt** [comment]
Thanks @mm326 I realised overnight that the node config override settings of the sequence identifier aren't being carried through. I think that's a minor addition though, so I propose to merge this and open a separate issue/pr to fix that.

File: ebu_tt_live/bindings/converters/ebutt1_ebutt3.py [hunk 1]
```diff
@@ -1,18 +1,260 @@
...
+        
+        if new_elem.documentMetadata and new_elem.documentMetadata.documentIdentifier is not None:
+            self._rememberDocumentIdentifier(new_elem.documentMetadata.documentIdentifier, dataset)
+
+        return new_elem
+
+    def _rememberDocumentIdentifier(self, documentIdentifier_in, dataset):
+        if 'documentIdentifier' in dataset:
+            raise Exception('Already got a documentIdentifier')
+        dataset['documentIdentifier']=documentIdentifier_in
```
Comment (danielthepope) [inline ebu_tt_live/bindings/converters/ebutt1_ebutt3.py]: And spaces around this `=` here too.

File: ebu_tt_live/documents/converters.py [hunk 2]
```diff
@@ -41,7 +42,9 @@ def ebutt1_to_ebutt3(ebutt1_in):
     :return:
     """
     converter = EBUTT1EBUTT3Converter()
-    ebutt3_bindings = converter.convert_document(ebutt1_in.binding)
+    doc_xml = ebutt1_in.get_xml()
+    ebutt1_doc = EBUTT1Document.create_from_xml(doc_xml)
+    ebutt3_bindings = converter.convert_document(ebutt1_doc.binding)
```
Comment (danielthepope) [inline ebu_tt_live/documents/converters.py]: This bit confuses me. Here we're taking an EBUTT1Document object, getting its XML, reading that back into an EBUTT1Document object, then converting it to EBUTT3.
The XML-and-back-again conversion seems unnecessary. Any particular reason for this?

File: ebu_tt_live/bindings/converters/ebutt1_ebutt3.py [hunk 3]
```diff
@@ -1,18 +1,260 @@
...
 log = logging.getLogger(__name__)
 
 
 class EBUTT1EBUTT3Converter(object):
 
-    def __init__(self):
+    _semantic_dataset = None
+    _sequenceIdentifier = None
+
+    def __init__(self, sequence_id):
```
Comment (danielthepope) [inline ebu_tt_live/bindings/converters/ebutt1_ebutt3.py]: By adding this extra `sequence_id` parameter, the converter's usage in `converters.py` won't work. Can you set a default value here?
Alternatively set a value in `converters.py`

**Comment by danielthepope** [comment]
Also worth mentioning that when this does get merged, it should be squashed. There are all sorts of duplicated commit messages that appear to change whitespace, and these should be removed from the history.

File: ebu_tt_live/documents/converters.py (hunk 2 above)
Comment (nigelmegitt) [reply ebu_tt_live/documents/converters.py]: It's consistent with the `ebutt3_to_ebuttd` function, but I agree it isn't clear why this approach is taken in either place. Also it looks to me as though neither function is ever even called.

File: ebu_tt_live/bindings/converters/ebutt1_ebutt3.py (hunk 3 above)
Comment (nigelmegitt) [reply ebu_tt_live/bindings/converters/ebutt1_ebutt3.py]: Yes, a value should be set in `converters.py` but I'm now wondering why that file even exists!

Comment (danielthepope) [reply ebu_tt_live/bindings/converters/ebutt1_ebutt3.py]: I was thinking we'd use those helper functions in Distiller - they seem pretty handy to me.

Comment (nigelmegitt) [reply ebu_tt_live/bindings/converters/ebutt1_ebutt3.py]: OK we haven't discussed exactly how we're going to instantiate the classes and run them in Distiller yet, so we should do that. My thinking was that we would be in closer control within the Distiller component of exactly how we do the conversion. For example wiring in the Denester etc.

File: ebu_tt_live/documents/ebutt1.py [hunk 4]
```diff
@@ -1,8 +1,10 @@
 from ebu_tt_live import bindings
-from ebu_tt_live.bindings import _ebuttlm as ebuttlm
+from ebu_tt_live.bindings import _ebuttlm as ebuttlm, _ebuttm as metadata
```
Comment (danielthepope) [inline ebu_tt_live/documents/ebutt1.py]: My linter suggests both these imports are unused. Can they be removed?

File: ebu_tt_live/documents/test/test_converters.py [hunk 5]
```diff
@@ -53,3 +55,109 @@ def test_ericsson_1(self):
...
+
+    def test_ericsson_smpte(self):
+
+        self.skipTest('SMPTE time conversion not supported yet')
+
+        xml_file = self._load_asset('converter_ericsson1_smpte.xml')
+
+        document = EBUTT1Document.create_from_xml(xml_file)
+        cdoc = ebutt1_to_ebutt3(document, sequence_id=self._seqId, 
+        use_doc_id_as_seq_id=True)
```
Comment (danielthepope) [inline ebu_tt_live/documents/test/test_converters.py]: The indentation of this line should be in line with the `(` of the line above

File: ebu_tt_live/documents/test/test_converters.py [hunk 6]
```diff
@@ -53,3 +55,109 @@ def test_ericsson_1(self):
...
+
+    def test_ericsson_media(self):
+
+        xml_file = self._load_asset('converter_ericsson1_media.xml')
+
+        document = EBUTT1Document.create_from_xml(xml_file)
+        cdoc = ebutt1_to_ebutt3(document, sequence_id=self._seqId, 
+        use_doc_id_as_seq_id=True)
```
Comment (danielthepope) [inline ebu_tt_live/documents/test/test_converters.py]: The indentation of this line should be in line with the `(` of the line above

File: ebu_tt_live/documents/test/test_converters.py [hunk 7]
```diff
@@ -47,9 +50,131 @@ def test_simple(self):
 
     def test_ericsson_1(self):
```
Comment (danielthepope) [inline ebu_tt_live/documents/test/test_converters.py]: Does the test name need to change now that you're using a different file name?

Comment (nigelmegitt) [reply ebu_tt_live/documents/test/test_converters.py]: sigh, I suppose it should

**Pull request merged**